### PR TITLE
[WIP] Project parameter

### PR DESCRIPTION
**Pull request by pocke**
Ref: https://skyarch.backlog.jp/view/NC-465

# Goal

CRUD for Project Paramter

- [x] model
- [x] migration
- [x] parameter index
- [x] update
- [x] destroy
  - [x] flag
- [x] create
- [x] I18n
- [x] authorize

# Next step

- Use a project parameter in CloudFormation, Cook, etc...

### CODE REVIEW

File: spec/models/project_parameter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+require_relative '../spec_helper'
+
+describe ProjectParameter, type: :model do
+  describe 'with validation' do
+    describe 'project_id and key' do
+      let(:project){create(:project)}
+      let(:key){'key'}
+      it 'should be uniq'do
+        param1 = create(:project_parameter, project: project, key: key)
```
Comment (skyarch-networks) [inline spec/models/project_parameter_spec.rb]: __[rubocop]__
* Lint/UselessAssignment: Useless assignment to variable - `param1`. Did you mean `param2`? [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: app/controllers/project_parameters_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  end
+
+  # Create / Update / Destroy 全部やる
+  # PUT /project_parameters
+  def update
+    # destroy されていない parameter の配列
+    parameters = JSON.parse(params.require(:params), symbolize_names: true)
+
+    # Update すべきな parameters
+    update_parameters = parameters.select{|x| x[:changed] && x[:id]}
```
Comment (skyarch-networks) [inline app/controllers/project_parameters_controller.rb]: __[rubocop]__
* Lint/UselessAssignment: Useless assignment to variable - `update_parameters`. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

**Comment by pocke** [comment]
@Joeper214 
please review and merge it :bow:

**Comment by Joeper214** [comment]
:+1: I'll review now.

**Comment by Joeper214** [comment]
same update process? 

`bundle install --path vendor/bundle`
`bower install`
```
$ sudo npm i -g gulp
$ cd frontend/
$ npm i
$ gulp tsd
$ gulp ts
$ cd ..
```

`bundle exec rake db:create`
`bundle exec rake db:migrate`

`bundle exec rake db:seed`

**Comment by pocke** [comment]
We only need the following process for update .

```sh
gulp ts
bundle exec rake db:migrate
```

**Comment by Joeper214** [comment]
LGTM!

**Pull request merged**